test/coarse_lib.cpp

```diff
@@ -114,7 +114,13 @@ INSTANTIATE_TEST_SUITE_P(
     CoarseWithMemoryStrategyTest, CoarseWithMemoryStrategyTest,
     ::testing::Values(UMF_COARSE_MEMORY_STRATEGY_FASTEST,
                       UMF_COARSE_MEMORY_STRATEGY_FASTEST_BUT_ONE,
-                      UMF_COARSE_MEMORY_STRATEGY_CHECK_ALL_SIZE));
+                      UMF_COARSE_MEMORY_STRATEGY_CHECK_ALL_SIZE),
+    ([](auto const &info) {
+        const char *names[] = {"UMF_COARSE_MEMORY_STRATEGY_FASTEST",
+                               "UMF_COARSE_MEMORY_STRATEGY_FASTEST_BUT_ONE",
+                               "UMF_COARSE_MEMORY_STRATEGY_CHECK_ALL_SIZE"};
+        return names[info.index];
+    }));
```

Comment:
Undefined behavior: I'm pretty sure that the automatic variable will be destroyed after the function return.
Please fix all occurrences.